cmd/namespace/sleep_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package namespace
+
+import (
+	"context"
+	"testing"
+
+	"github.com/okteto/okteto/internal/test/client"
+	"github.com/okteto/okteto/pkg/okteto"
+	"github.com/okteto/okteto/pkg/types"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func Test_SleepNamespace(t *testing.T) {
+	ctx := context.Background()
+	currentNamespace := "current"
+	usr := &types.User{
+		Token: "test-token",
+	}
+	initNamespaces := []types.Namespace{
+		{
+			ID: currentNamespace,
+		},
+		{
+			ID: "test-1",
+		},
+	}
+	var tests = []struct {
+		name string
+		// toSleepNs the namespace to sleep
+		toSleepNs                       string
+		initialNamespacesAtOktetoClient []types.Namespace
+		fakeOkClient                    *client.FakeOktetoClient
+		fakeK8sClient                   *fake.Clientset
+		err                             error
+	}{
+		{
+			name:                            "sleeps existing ns, the current one",
+			toSleepNs:                       currentNamespace,
+			initialNamespacesAtOktetoClient: initNamespaces,
+			fakeOkClient: &client.FakeOktetoClient{
+				Namespace:    client.NewFakeNamespaceClient(initNamespaces, nil),
+				Users:        client.NewFakeUsersClient(usr),
+				StreamClient: client.NewFakeStreamClient(&client.FakeStreamResponse{}),
+			},
+			fakeK8sClient: fake.NewSimpleClientset(),
+		},
+		{
+			name:                            "sleeps existing ns, not the current one",
+			toSleepNs:                       "test-1",
+			initialNamespacesAtOktetoClient: initNamespaces,
+			fakeOkClient: &client.FakeOktetoClient{
+				Namespace:    client.NewFakeNamespaceClient(initNamespaces, nil),
+				Users:        client.NewFakeUsersClient(usr),
+				StreamClient: client.NewFakeStreamClient(&client.FakeStreamResponse{}),
+			},
+			fakeK8sClient: fake.NewSimpleClientset(),
+		},
+		{
+			name:                            "sleep non existing ns",
```

Comment:
Hey @jLopezbarb @pchico83!
In this case, it expects an error to be returned as there is no `test-non-existing` namespace present but while trying to run this test, it still sleeps a `test-non-existing` namespace, idk why. Can you help me out here?
![Screenshot from 2023-05-10 00-09-24](https://github.com/okteto/okteto/assets/98955085/47033fa3-564d-4935-9492-bcc4c9399836)